### ISSUE TITLE
Table accessibility

### ISSUE THREAD

**Issue by ajanec01** [craftcms/cms#7445]
### Overview of accessibility issues in the table element- Assets and Categories page

- Row headers would benefit from the `scope` attribute 
- First column header with a checkbox not announced by the screen reader (checked with orca); even if it were announced then the “select all” label would not provide much context; consider a `colspan` with a visually hidden text
![column-header-with-aria-label](https://user-images.githubusercontent.com/47136246/105914051-b31cd600-6025-11eb-8279-9a3a9f54ecf8.png)
- I’ve been getting incorrect results for column headers and for the row headers depending on which browser I use. (Firefofox/ Chrome); I think it’s because the row headers are in the second column; I’ve been using Orca so it needs to be double-checked 
- Checkboxes in the first column (main table)  are not keyboard accessible
- Not possible to sort cells by column headers (using keyboard only)- they are also missing aria-sort; similar solution in example 2 of [aria grids](https://www.w3.org/TR/wai-aria-practices-1.2/examples/grid/dataGrids.html) 
- Status update for the number of search results not conveyed to sr users- consider `aria-live=”polite”`
![status-update](https://user-images.githubusercontent.com/47136246/104253989-b92f8600-546d-11eb-848d-704aab4ba49f.png)
- Export button needs `aria-expanded` and `aria-controls`
![missing-state](https://user-images.githubusercontent.com/47136246/104254031-d8c6ae80-546d-11eb-82c6-f1264ee036bb.png)

### Additional info

- Craft version: 3.5.18

**Comment by ajanec01** [comment]
The same issues also apply to the Users page

**Comment by gcamacho079** [comment]
All of these issues have been resolved.
